### PR TITLE
Add a sanity check for GPU pipeline

### PR DESCRIPTION
**Pull request by trontrytel**
Add a tests agains this issue: https://github.com/CliMA/SurfaceFluxes.jl/issues/142

### CODE REVIEW

File: test/gpu_tests.jl [hunk 1]
```diff
@@ -121,6 +121,28 @@ end
     end
 end
 
+@kernel function test_1_moment_micro_smooth_transition_kernel!(
+    acnv1M,
+    output::AbstractArray{FT},
+    ql
+) where {FT}
+
+    i = @index(Group, Linear)
```
Comment (charleskawczynski) [inline test/gpu_tests.jl:130]: I think we need to use `Global` here? xref: https://github.com/CliMA/SurfaceFluxes.jl/issues/142

Comment (trontrytel) [reply test/gpu_tests.jl:130]: Thank you! I wonder how the tests are passing with that. 

I'll fix it following what you did in SurfaceFluxes

File: test/gpu_tests.jl [hunk 2]
```diff
@@ -747,6 +761,19 @@ function test_gpu(FT)
...
+        (; output, ndrange) = setup_output(dims, FT)
+        ql = ArrayType([FT(1e-3), FT(5e-4)])
+
+        kernel! =
+            test_1_moment_micro_smooth_transition_kernel!(backend, work_groups)
+        kernel!(rain.acnv1M, output, ql; ndrange)
+
+        # Sanity checks for the GPU KernelAbstractions workflow
+        # See https://github.com/CliMA/SurfaceFluxes.jl/issues/142
+        @test !any(isequal(Array(output), -99999.99))
```
Comment (charleskawczynski) [inline test/gpu_tests.jl:774]: Is `output` filled with `-99999.99` when its initialized?

Comment (charleskawczynski) [reply test/gpu_tests.jl:774]: If so, looks good!

**Pull request merged**